### PR TITLE
feat: setup mirror

### PR DESCRIPTION
**Pull request by nuanyang233**
## Description

Closes #MF-2035

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/background/services/site-adaptors/connect.ts [hunk 1]
```diff
@@ -37,6 +37,7 @@ export async function connectSite(
     profile?: ProfileIdentifier,
 ) {
     const worker = definedSiteAdaptors.get(network)
+    console.log(definedSiteAdaptors)
```
Comment (guanbinrui) [inline packages/mask/background/services/site-adaptors/connect.ts]:
```suggestion
```

File: packages/mask/src/social-network-adaptor/mirror.xyz/collecting/identity.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    const assign = async () => {
+        await delay(2000)
+
+        const writer = await getCurrentUserInfo()
+        if (!writer) return
+
+        ref.value = {
+            avatar: writer.avatarURL,
+            nickname: writer.displayName,
+            identifier: ProfileIdentifier.of(mirrorBase.networkIdentifier, writer.address).unwrapOr(undefined),
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/mirror.xyz/collecting/identity.ts:31]: Lost `bio` and `homepage` here.

Comment (nuanyang233) [reply packages/mask/src/social-network-adaptor/mirror.xyz/collecting/identity.ts:29]: Mirror doesn't seem to have `bio`.

File: packages/mask/src/social-network-adaptor/mirror.xyz/collecting/identity.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    cancel: AbortSignal,
+) {
+    const assign = async () => {
+        const script = document.getElementById('__NEXT_DATA__')?.innerHTML
+        if (!script) return
+        const INIT_DATA = JSON.parse(script)
+        if (!INIT_DATA) return
+        const writer = INIT_DATA.props?.pageProps?.project as Writer
+        if (!writer) {
+            if (location.pathname !== '/dashboard') return
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/mirror.xyz/collecting/identity.ts]: `location.pathname.includes('/dashboard')`.

File: packages/mask/src/social-network-adaptor/mirror.xyz/collecting/identity.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+            // when current page is dashboard
+            const currentUser = await getCurrentUserInfo()
+            if (!currentUser) return
+            ref.value = {
+                avatar: currentUser?.avatarURL,
+                nickname: currentUser?.displayName,
+                identifier: ProfileIdentifier.of(mirrorBase.networkIdentifier, currentUser.address).unwrapOr(undefined),
+            }
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/mirror.xyz/collecting/identity.ts:64]: Did you forget `return` here?
```suggestion
            }
            return
```

**Review by guanbinrui — APPROVED** [review]

File: packages/mask/background/network/gun/encryption/queryPostKey.ts [hunk 5]
```diff
@@ -343,6 +344,7 @@ namespace Version37 {
         if (x === SocialNetworkEnum.Twitter) return '37-tw'
         if (x === SocialNetworkEnum.Minds) return '37-minds'
         if (x === SocialNetworkEnum.Instagram) return '37-ins'
+        if (x === SocialNetworkEnum.Mirror) return '37-mirror'
```
Comment (Jack-Works) [inline packages/mask/background/network/gun/encryption/queryPostKey.ts:347]: If this website does not support sending posts, please do not add it (like OpenSea doesn't show here.)

File: packages/mask/src/utils/fetcher.ts [hunk 6]
```diff
@@ -35,6 +36,8 @@ function canAccessAsContent(url: string) {
         ['https://abs.twimg.com', 'https://upload.twitter.com'].includes(target.origin)
     )
         return true
+
+    if (location.origin.endsWith(EnhanceableSite.Mirror)) return true
```
Comment (Jack-Works) [inline packages/mask/src/utils/fetcher.ts:40]: this does not make sense. do you mean: `url.endsWith(...) && location.origin.endsWith(...)`?

**Pull request merged**